### ISSUE TITLE
[feat] CLN switch the pay plugin to renepay

### ISSUE THREAD

**Issue by openoms** [raspiblitz/raspiblitz#4777]
Rationale: the pay plugin cannot deliver some payments where renepay is very good.

It can be done with a simple translator plugin taking over the pay RPC hook:

https://gist.github.com/openoms/8626623a124ff1164096f9b8d971195c

Tested so far with paying from 
- Zeus via CLNrest
- boltcard via LNbits

On renepay: https://blog.blockstream.com/core-lightning-v24-08-steel-backed-up-channels/

**Comment by surfac3** [comment]
I tested this as well using both cmd line and Zeus (CLNrest) with CLN v24.08-60 on RaspiBlitz v1.11.2 

Payments ran smoothly with smaller amount(500k) and larger (1mil).

**Comment by steepdawn974** [comment]
Works like a charm :rocket: 
Tested via Zeus CLNRest over Tor

```
2024-10-04T11:46:51.529Z INFO    plugin-cl-plugin.pay-hook-to-renepay.py: Intercepted 'pay' command, redirecting to 'renepay'
2024-10-04T11:46:51.529Z INFO    plugin-cl-plugin.pay-hook-to-renepay.py: Original params: {'bolt11': 'LNBC..., 'maxfeepercent': '5.0'}
2024-10-04T11:46:51.529Z INFO    plugin-cl-plugin.pay-hook-to-renepay.py: Parameter 'maxfeepercent' not used by 'renepay', ignoring.
2024-10-04T11:46:51.529Z INFO    plugin-cl-plugin.pay-hook-to-renepay.py: Adjusted params: {'invstring': 'LNBC....'}
2024-10-04T11:46:53.044Z INFO    lightningd: Sending xxxxx000msat over 3 hops to deliver xxxxx000msat
....
```

**Comment by chrisguida** [comment]
There is also an "official" version of this plugin, for those interested: https://github.com/Lagrang3/default-renepay

**Comment by openoms** [comment]
> There is also an "official" version of this plugin, for those interested: https://github.com/Lagrang3/default-renepay

Yes, it looks it does exactly the same. Forwards the `pay` RPC hook to `renepay`and matches the parameters. Both are quite simple, less than 100 lines in python. Feel free to test both.

**Comment by openoms** [comment]
when sending with Zeus to a BTCPay lightningaddress getting the error with both plugins:
```
INFO    plugin-clnrest: RPC Error: {'code': -32602, 'message': 'bolt11 uses description_hash, but you did not provide description parameter'}
```

**Comment by openoms** [comment]
The new CLN release will have this sorted:

>xpay is a new, experimental plugin for payments. It's rewritten from the ground up, on top of another plugin called askrene, which provides advanced routing advice for payments. The two play together: xpay tells askrene about the results of payment attempts, so it learns over time. The hard work was done by @Lagrang3 who wrote and refined the minimum-cost-flow solver which is the heart of askrene.
For reckless souls, xpay-handle-pay can be set to true (even at runtime, using setconfig) to have xpay take over the simple variants of pay.

Will close this issue after testing.

**Comment by tobomobo** [comment]
I am testing `xpay` and `CLN v24.11` as we speak, and paying a bolt11 invoice in the CLI works flawlessly. 

With Zeus, however, not so much.

[https://github.com/ElementsProject/lightning/issues/7926](https://github.com/ElementsProject/lightning/issues/7926)

**Comment by chrisguida** [comment]
@tobomobo

1) this issue is about `renepay`, not `xpay`
2) you need to set `xpay-handle-pay` as suggested here if you want `xpay` to take over for `pay`: https://github.com/ElementsProject/lightning/releases/tag/v24.11

**Comment by tobomobo** [comment]
> @tobomobo
> 
> 1. this issue is about `renepay`, not `xpay`
> 2. you need to set `xpay-handle-pay` as suggested here if you want `xpay` to take over for `pay`: https://github.com/ElementsProject/lightning/releases/tag/v24.11

Hi, yes the issue itself is/was. 

I was referring to [Openoms](https://github.com/raspiblitz/raspiblitz/issues/4777?notification_referrer_id=NT_kwDOA3HyirQxMjcxMjk4MzMyNzo1Nzc5OTMwNg#issuecomment-2507147380). 

I have have set `xpay-handle-pay` to true

**Comment by chrisguida** [comment]
@tobomobo ah, apologies. carry on.

**Comment by rootzoll** [comment]
@openoms we are shipping RaspiBlitz v1.12.0 with Core Lightning v25.02.2 or above, so we can close this issue, correct?

**Comment by geco91** [comment]
@openoms any todo for v1.12.0 here or push to later?

**Comment by openoms** [comment]
xpay has been intruduced which does similar so closing this.
